src/current/v23.2/column-level-encryption.md

```diff
@@ -0,0 +1,200 @@
+{{site.data.alerts.end}}
+
+## Security considerations
+
+### Whether to use `encrypt` and `decrypt` or `encrypt_iv` and `decrypt_iv`?
+
+Both `encrypt` and `decrypt` have `*_iv` variants: `encrypt_iv` and `decrypt_iv`.
+
+The benefits of using `encrypt_iv` and `decrypt_iv` include:
+
+- Avoid Repetition: If you use the same encryption key and method to encrypt the same plaintext multiple times without an IV, you'll get the same ciphertext every time. This repetition can provide a point of attack for someone trying to break the encryption. By using an IV, even the same plaintext will produce different ciphertexts, provided a different IV is used each time.
+- Defend Against Pattern Analysis: Without an IV, if two users have the same SSN or piece of data, their encrypted values will also be the same. An attacker can exploit these patterns. By using different IVs for each encryption, the encrypted values will be different even if the plaintext values are the same.
+- Cipher Block Chaining (CBC) Mode: Many encryption algorithms, like AES, operate on blocks of data. In modes like CBC, the previous block of ciphertext is used as an IV for the encryption of the next block. This means that even if there are patterns in the plaintext, they won't appear in the ciphertext. However, for the first block, there is no previous block of ciphertext, so an IV is used. This is another way IVs help in breaking up patterns in the ciphertext.
+- Mitigate Replay Attacks: Since the IV is typically random and changed for every encryption, it makes replay attacks more difficult. An attacker can't simply take an old piece of encrypted data and send it again, as the IV will likely have changed.
+
+The drawbacks of using `encrypt_iv` and `decrypt_iv` include:
+
+- Storage: You need to store the IV alongside the ciphertext. It's common practice to prepend or append the IV to the ciphertext before storing it. Unlike the encryption key, the IV doesn't need to be kept secret, but it does need to be known for decryption.
+- Randomness: It's crucial that IVs are random and not predictable. If an attacker can predict the next IV, some of the security benefits are negated.
+- Unique IVs with the Same Key: While IVs need to be random, it's also essential that the same IV isn't used twice with the same encryption key. Doing so can leak information about the plaintext.
+
+You will need to assess your risk profile to determine which functions to use.
+
+### How AES variants are determined
+
+The actual AES variant (AES-128, AES-192, or AES-256) is determined by the length of the encryption key you provide in the functions:
+
+- AES-128: 16-byte key
+- AES-192: 24-byte key
+- AES-256: 32-byte key
+
+Keys have to be 16-, 24-, or 32-byte lengths and map to the corresponding AES encryption strength.
+
+## Performance considerations
```

Comment:
Use of encrypt built-in can have anywhere from 10-40% overhead depending on the length of the data being encrypted and the hardware provisioned for CockroachDB. We measured baseline performance in a 3 node CockroachDB cluster running on three n1-standard-4 nodes.
```
-- Without encrypt/decrypt - 60-80 ms total
with a as (select lpad((1000 * random())::int::string, 3, '0') || '-' || lpad((100 * random())::int::string, 2, '0') || '-' || lpad((10000 * random())::int::string, 4, '0') ssn from generate_series(1, 1.0E+04)) select ssn from a;

-- With encrypt/decrypt - 80-100ms
with a as (select lpad((1000 * random())::int::string, 3, '0') || '-' || lpad((100 * random())::int::string, 2, '0') || '-' || lpad((10000 * random())::int::string, 4, '0') ssn from generate_series(1, 1.0E+04)) select convert_from(decrypt(encrypt(ssn::bytea, '\xd54a43d2a4caf8d3fbe4e4f711b39d4a0fedf26ac0dcdfb0811c2078a6a9cd147e77da38e35e14cacfc79c7e11a052c4bc9449e1d6fa280dcdc45bb4004f1648', 'aes'), '\xd54a43d2a4caf8d3fbe4e4f711b39d4a0fedf26ac0dcdfb0811c2078a6a9cd147e77da38e35e14cacfc79c7e11a052c4bc9449e1d6fa280dcdc45bb4004f1648', 'aes'), 'UTF8') from a;

-- With encrypt only - 80-100ms
with a as (select lpad((1000 * random())::int::string, 3, '0') || '-' || lpad((100 * random())::int::string, 2, '0') || '-' || lpad((10000 * random())::int::string, 4, '0') ssn from generate_series(1, 1.0E+04)) select encrypt(ssn::bytea, '\xd54a43d2a4caf8d3fbe4e4f711b39d4a0fedf26ac0dcdfb0811c2078a6a9cd147e77da38e35e14cacfc79c7e11a052c4bc9449e1d6fa280dcdc45bb4004f1648', 'aes') from a;
```

It is important to do benchmark these built in for your particular CockroachDB setup to establish the performance implications.

Reply:
@mgoddard borrowed these from your testing and did some testing on roachprod. Let us know if this section makes sense in terms of establishing performance implications.

Reply:
thanks @dikshant I've added this info (slightly edited) to the doc.  It should be ready for review now, I'll add you and is it @andyyang890 ?